trampoline/build-trampoline.py

```diff
@@ -0,0 +1,58 @@
+import subprocess
+from pathlib import Path
+import argparse
+
+
+def get_default_target() -> str:
+    result = subprocess.run(["rustc", "-vV"], capture_output=True, text=True, check=True)
+    for line in result.stdout.splitlines():
+        if line.startswith("host:"):
+            return line.split(":")[1].strip()
+    raise RuntimeError("Unable to determine default target")
+
+
+def build_trampoline_binary(target: str, target_dir: Path) -> None:
+    subprocess.run(
+        [
+            "cargo",
+            "build",
+            "--release",
+            "--target",
+            target,
+            "--target-dir",
+            target_dir,
+            "--manifest-path",
+            "trampoline/Cargo.toml",
+        ],
+        check=True,
+    )
+
+
+def compress_binary(target: str, target_dir: Path) -> None:
```

Comment:
I would rename it a little
`target_dir` -> `binary_dir`
idea is that initally I though that it is something configurable, but it is the dir from where we load binaries but we always place compressed binaries inside `trampoline/binaries`